src/error.rs

```diff
@@ -262,6 +269,85 @@ fn from_sslerror_ref(err: &ssl::Error) -> Error {
     }
 }
 
+#[cfg(feature = "snappy")]
+fn from_snap_error_ref(err: &::snap::Error) -> Error {
+    match err {
+        &::snap::Error::TooBig { given, max } => {
```

Comment:
`error_chain` sounds good to me 👍 i think the benefits could go beyond just dealing with the snappy errors. i see at least openssl and io errors being wrapped here. `Box<Error>` sounds like an OK alternative. however, since `kafka-rust` is a library i wouldn't accept the `String` approach.